src/coreclr/src/jit/simd.cpp

```diff
@@ -765,6 +765,15 @@ var_types Compiler::getBaseTypeAndSizeOfSIMDType(CORINFO_CLASS_HANDLE typeHnd, u
             }
         }
 
+#if defined(TARGET_XARCH)
+        // Even though Vector256 is TYP_SIMD32, if AVX isn't supported, then it must
+        // be treated as a regular struct
+        if (size == YMM_REGSIZE_BYTES && (simdBaseType != TYP_UNKNOWN) && !compExactlyDependsOn(InstructionSet_AVX))
+        {
+            simdBaseType = TYP_UNKNOWN;
+        }
+#endif // TARGET_XARCH
+
```

Comment:
Seems odd that we didn't need this before but need this now -- is it just for the R2R case because you modified `largestEnregisterableStructSize` ?

Reply:
Yes, this only happens in the R2R case, and only because of my trickery in `largestEnregisterableStructSize`. It has the nice property of making the getBaseTypeAndSizeOfSIMDType safe to call at all times, which I thought was a nice upgrade.